Force fp16 for vision_maskrcnn inference

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113110

For fp16 for maskrcnn inference (doesnt support bf16). Also skip phi_1_5 in training - it OOMs even with batch size 1


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

eellison committed:
Force fp16 for vision_maskrcnn inference

[ghstack-poisoned]

eellison committed:
Update on "Force fp16 for vision_maskrcnn inference"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

eellison committed:
Update on "Force fp16 for vision_maskrcnn inference"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

xmfan commented on benchmarks/dynamo/common.py:
```diff
@@ -2131,6 +2135,12 @@ def cast_based_on_args(self, model, example_inputs):
                 self.setup_amp()
+            elif self.args.only in self.force_fp16_for_bf16_models:
+                log.warning(
+                    "Model %s does not support float16, running with bfloat16 instead",
```
```suggestion
                    "Model %s does not support bfloat16, running with float16 instead",
```

xmfan replied:
also, when should we force to fp16 and when should we force to amp (code above this condition)

eellison replied:
Sometimes models need more precision than fp16/bf16, so we run in amp. In this case the kernel does not support bf16

eellison committed:
Update on "Force fp16 for vision_maskrcnn inference"


For fp16 for maskrcnn inference (doesnt support bf16). Also skip phi_1_5 in training - it OOMs even with batch size 1


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison commented:
@pytorchbot merge

eellison committed:
Update on "Force fp16 for vision_maskrcnn inference"


For fp16 for maskrcnn inference (doesnt support bf16). Also skip phi_1_5 in training - it OOMs even with batch size 1


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison commented:
@pytorchbot merge
